hr_attendance_management/models/hr_attendance_day.py

```diff
@@ -431,19 +431,45 @@ def create(self, vals):
     @api.multi
     def write(self, vals):
         res = super(HrAttendanceDay, self).write(vals)
-        self.recompute_period_if_old_day()
+        if 'paid_hours' in vals:  # TODO check
+            self.recompute_period_if_old_day()
         return res
 
     ##########################################################################
     #                             PUBLIC METHODS                             #
     ##########################################################################
     @api.multi
     def recompute_period_if_old_day(self):
-        last_period = self.env['base.config.settings'].create({}) \
-            .get_last_balance_cron_execution()
         for day in self:
-            if day.date < last_period:
-                day.employee_id._update_past_period_balance()
+            lower_bound_period = self.env['hr.employee.period'].search([
+                ('employee_id', '=', day.employee_id.id),
+                ('end_date', '<=', day.date)
+            ], order='end_date desc', limit=1)
```

Comment:
I can't use that because I don't need the first or last period of the employee but the one just before/after the period of the current day. I will refactor a bit to gain in speed